Allow multiple (or no) Closures on a flow channel

joshuahansel opened the pull request:
Closes #29594

joshuahansel committed:
Allowed multiple Closures to be supplied

Refs #29594

joshuahansel committed:
Added FunctorClosures and demonstrated using multiple closures

Closes #29594

licharlot reviewed (commented):
I'd suggest adding a test where no closures are provided to the flow channel.

joshuahansel commented on modules/thermal_hydraulics/src/components/FlowChannelBase.C:
```diff
@@ -89,7 +89,11 @@ FlowChannelBase::validParams()
+      "closures",
+      {},
+      "Closures object(s). This is optional since closure relations can be supplied directly by "
+      "Materials as well.");
```
I think it should be deprecated, but I need to think about it how. It currently has the ability to add wall temperature material property, so it's not actually none. I'll probably make a new closures like `WallTemperature1PhaseClosures`

licharlot replied:
That could be an option, but asking the user to create the material is also an option.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2652082/), step Docs: sync website on 4e2a3a6 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/29752/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2652129/), step Generate coverage on 4e2a3a6 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>a6d500</th>
      <th colspan=4>#29752 4e2a3a</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.25%</td>
      <td>85.17%</td>
      <td>-0.08%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>108135</td>
      <td>108039</td>
      <td>-96</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18715</td>
      <td>18811</td>
      <td>+96</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/29752/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/29752/coverage/framework)


## Modules coverage

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>a6d500</th>
      <th colspan=4>#29752 4e2a3a</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>88.38%</td>
      <td>88.38%</td>
      <td>-0.00%</td>
      <td>95.65%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>13906</td>
      <td>13936</td>
      <td>+30</td>
      <td>44</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1828</td>
      <td>1832</td>
      <td>+4</td>
      <td>2</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/29752/coverage_diff/thermal_hydraulics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/29752/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/29752/coverage/xfem)

</details>



This comment will be updated on new commits.

joshuahansel committed:
Removed unnecessary material property from T_avg material

The average wall temperature material required an average wall
heat transfer coefficient material, but this quantity can be
computed directly from the other parameters, so the parameter
was removed to ease requirements of the material.

Refs #29594

joshuahansel committed:
Added WallTemperature1PhaseClosures

Refs #29594

joshuahansel committed:
Deprecated Closures1PhaseNone

Refs #29594

joshuahansel committed:
Restricted test to require air submodule

Refs #29594
